### ISSUE TITLE
[BUG] pod evict failed with wrong eviction api

### ISSUE THREAD

**Issue by j4ckstraw** [koordinator-sh/koordinator#1373]
**What happened**:
```
E0612 15:17:16.770432  790193 resmanager.go:203] evict pod default/be-7449f95c6f-5qlhl failed, reason: EvictPodByBECPUSatisfaction, error: Eviction in version "v1" cannot be handled as a Eviction: no kind "Eviction" is registered for version "policy/v1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"
```

**What you expected to happen**:
evict return success

**How to reproduce it (as minimally and precisely as possible)**:
1. deploy a be pod with batch-cpu: 4k, run `stress-ng -c 4` in pod
2. stress node to trigger pod evict by `stress-ng`

**Anything else we need to know?**:

**Environment**:
- App version: koordinator 1.2.0
- Kubernetes version (use `kubectl version`): 1.21.3
- Install details (e.g. helm install args):
- Node environment (for koordlet/runtime-proxy issue):
  - Containerd/Docker version:
  - OS version: Centos7
  - Kernal version: 4.19.91
  - Cgroup driver: systemd
- Others:

**Comment by j4ckstraw** [comment]
I think I found the root cause.
policy/v1 Eviction is available in v1.22+ reference to https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/#calling-the-eviction-api, our cluster is v1.21 which is not support eviction api v1. 
But 

evict in koordinator https://github.com/koordinator-sh/koordinator/blob/7798f3a5daf6ced33217f17665b5833a5988f3b2/pkg/util/pod_evict.go#L40

compare with the kubectl drain command logic
https://github.com/kubernetes/kubernetes/blob/18d05b646d09b2971dc5400bc288062b0414e8cf/staging/src/k8s.io/kubectl/pkg/drain/drain.go#L141

**Comment by j4ckstraw** [comment]
policy/v1 Eviction is available in v1.22+ reference to https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/#calling-the-eviction-api, while our cluster is v1.21 which is not support eviction api v1. 
but koordinator use eviction v1 to evict pod

**Comment by j4ckstraw** [comment]
I found kubectl drain use `d.Client.PolicyV1().Evictions(eviction.Namespace).Evict(context.TODO(), eviction)`
https://github.com/kubernetes/kubernetes/blob/18d05b646d09b2971dc5400bc288062b0414e8cf/staging/src/k8s.io/kubectl/pkg/drain/drain.go#L141
what's the difference to koordinator `kubernetes.CoreV1().Pods(namespace).EvictV1beta1(ctx, &policyv1beta1.Eviction{})` https://github.com/koordinator-sh/koordinator/blob/7798f3a5daf6ced33217f17665b5833a5988f3b2/pkg/util/pod_evict.go#L42

**Comment by eahydra** [comment]
fixed by #1393
